### PR TITLE
POL-172 Update AWS policies for credential Service 2

### PR DESCRIPTION
**Pull request by KTebrewal**
### Description

Update listed AWS policies to support the credential service.  The changes include-
-Replace auth declaration with the credential declaration.
-Remove long_description field and use new info hash for Version, providers and service.
-Update tenant to single.
-Update readme to replace credential instructions with instructions to use credential service.
-Update CWF definitions as needed. 

### Issues Resolved

Changed the authentication to credential services.

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### Running template Link

AWS Savings Plan Recommendations - https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e144c8e372f7a0001f736df

image

AWS Subnet Name Tag Sync-
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e26aa1706101000014d560e

AWS Unencrypted ELB Listeners (ALB/NLB)-
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e26abe006101000014d5610

AWS Unencrypted ELB Listeners (CLB)-
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e26ae10e4ca220001b69e3d

AWS Unencrypted RDS Instances - https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e203d8f32401200016e02f3
image

AWS Unencrypted Volumes - https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e26af244ec7b300014e66c0

image

AWS Unused ECS Clusters-
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e26a1eb4ec7b300014e66a7
image

AWS VPC Name Tag Sync-
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e26c338e4ca220001b69e51

AWS Unencrypted S3 Buckets-
https://governance.rightscale.com/org/1105/projects/60073/policy-incidents/5e26d01a4ec7b300014e66ed
image

### CODE REVIEW

File: compliance/aws/ecs_unused/AWS_Unused_ECS_Clusters.pt [hunk 1]
```diff
@@ -5,6 +5,13 @@ short_description "Report and remediate any ECS clusters that are not currently
 long_description "Version: 1.2"
 category "Compliance"
 severity "low"
+tenancy "single"
```
Comment (cdwilhelm) [inline compliance/aws/ecs_unused/AWS_Unused_ECS_Clusters.pt]: tenancy can be removed

File: operational/aws/subnet_name_sync/README.md [hunk 2]
```diff
@@ -1,39 +1,37 @@
-## AWS Subnet Name Tag Sync Policy Template
+# AWS Subnet Name Tag Sync Policy Template
```
Comment (cdwilhelm) [inline operational/aws/subnet_name_sync/README.md]: Remove Policy Template from Name

File: operational/aws/subnet_name_sync/README.md [hunk 3]
```diff
@@ -1,39 +1,37 @@
...
+
 - Synchronizes AWS Subnet names to Subnets in Cloud Management
 
-### Input Parameters
+## Input Parameters
 
 This policy has the following input parameter required when launching the policy.
 
-- *Email addresses of the recipients you wish to notify* - A list of email addresses to notify
+- Email addresses of the recipients you wish to notify* - A list of email addresses to notify
```
Comment (cdwilhelm) [inline operational/aws/subnet_name_sync/README.md]: Missing leading *

File: operational/aws/subnet_name_sync/aws_subnet_name_sync.pt [hunk 4]
```diff
@@ -2,9 +2,16 @@ name "AWS Subnet Name Tag Sync"
 rs_pt_ver 20180301
 type "policy"
 short_description "Ensures a Subnet name in Cloud Management reflect the value of the Subnet name tag in AWS. See the [README](https://github.com/rightscale/policy_templates/tree/master/operational/aws/subnet_name_sync) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
-long_description "Version: 1.3"
+long_description ""
 severity "medium"
 category "Operational"
+tenancy "single"
```
Comment (cdwilhelm) [inline operational/aws/subnet_name_sync/aws_subnet_name_sync.pt]: remove tenancy

File: operational/aws/subnet_name_sync/aws_subnet_name_sync.pt [hunk 5]
```diff
@@ -33,11 +40,11 @@ end
...
 
-auth "auth_aws", type: "aws" do
-  version 4
-  service 'ec2'
-  access_key cred('AWS_ACCESS_KEY_ID')
-  secret_key cred('AWS_SECRET_ACCESS_KEY')
+credentials "auth_aws" do
+  schemes "aws"
+  label "AWS"
+  description "Select the AWS IAM Credential from the list"
```
Comment (cdwilhelm) [inline operational/aws/subnet_name_sync/aws_subnet_name_sync.pt]: remove IAM.  
The credentail can use the AWS IAM or STS credentials.

File: operational/aws/vpc_name_sync/README.md [hunk 6]
```diff
@@ -1,40 +1,37 @@
-## AWS VPC Name Tag Sync Policy Template
+# AWS VPC Name Tag Sync Policy Template
```
Comment (cdwilhelm) [inline operational/aws/vpc_name_sync/README.md]: please remove policy template

File: security/aws/clb_unencrypted/README.md [hunk 7]
```diff
@@ -1,27 +1,32 @@
...
- 
-### Required RightScale Roles
- 
-- admin or credential_viewer
 
-### AWS Required Permissions
+## Input Parameters
+
+- Email addresses of the recipients you wish to notify* - A list of email addresses to notify
+- Ignore tags* - CLB with any of these tags will be ignored
```
Comment (cdwilhelm) [inline security/aws/clb_unencrypted/README.md]: Missing the leading *

File: security/aws/clb_unencrypted/aws_clb_encryption.pt [hunk 8]
```diff
@@ -1,10 +1,17 @@
 name "AWS Unencrypted ELB Listeners (CLB)"
 rs_pt_ver 20180301
 type "policy"
-short_description "Report any AWS Classic Load Balancers w/Internet-facing Unencrypted Listeners. \n See the [README](https://github.com/rightscale/policy_templates/tree/master/security/aws/clb_unencrypted) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
-long_description "Version: 1.0"
+short_description "Report any AWS Classic Load Balancers w/Internet-facing Unencrypted Listeners. \n See the [README](https://github.com/rightscale/policy_templates/tree/master/security/aws/clb_unencrypted) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
+long_description ""
 category "Security"
 severity "medium"
+tenancy "single"
```
Comment (cdwilhelm) [inline security/aws/clb_unencrypted/aws_clb_encryption.pt]: remove tenancy

File: security/aws/clb_unencrypted/aws_clb_encryption.pt [hunk 9]
```diff
@@ -35,11 +42,11 @@ end
...
 
-auth "auth_aws", type: "aws" do
-  version 4
-  service "elasticloadbalancing"
-  access_key cred('AWS_ACCESS_KEY_ID')
-  secret_key cred('AWS_SECRET_ACCESS_KEY')
+credentials "auth_aws" do
+  schemes "aws"
+  label "AWS"
+  description "Select the AWS IAM Credential from the list"
```
Comment (cdwilhelm) [inline security/aws/clb_unencrypted/aws_clb_encryption.pt]: Remove IAM

File: security/aws/ebs_unencrypted_volumes/AWS_Unencrypted_Volumes.pt [hunk 10]
```diff
@@ -5,6 +5,14 @@ short_description "Report any Elastic Block Store (EBS) volumes in AWS that are
 long_description "Version: 1.1"
 category "Security"
 severity "low"
+tenancy "single"
```
Comment (cdwilhelm) [inline security/aws/ebs_unencrypted_volumes/AWS_Unencrypted_Volumes.pt]: remove tenancy

File: security/aws/ebs_unencrypted_volumes/AWS_Unencrypted_Volumes.pt [hunk 11]
```diff
@@ -35,11 +43,11 @@ end
...
 
-auth "auth_aws", type: "aws" do
-  version "4"
-  service "ec2"
-  access_key cred("AWS_ACCESS_KEY_ID")
-  secret_key cred("AWS_SECRET_ACCESS_KEY")
+credentials "auth_aws" do
+  schemes "aws"
+  label "AWS"
+  description "Select the AWS IAM Credential from the list"
```
Comment (cdwilhelm) [inline security/aws/ebs_unencrypted_volumes/AWS_Unencrypted_Volumes.pt]: remove IAM

File: security/aws/ebs_unencrypted_volumes/README.md [hunk 12]
```diff
@@ -1,26 +1,31 @@
...
- 
-- policy_manager
-- admin or credential_viewer
-
-### AWS Required Permissions
+
+## Input Parameters
+
+- Email addresses of the recipients you wish to notify* - A list of email addresses to notify
+- Ignore tags* - EBS volumes with any of these tags will be ignored
```
Comment (cdwilhelm) [inline security/aws/ebs_unencrypted_volumes/README.md]: Missing the leading *

File: security/aws/elb_unencrypted/README.md [hunk 13]
```diff
@@ -1,27 +1,32 @@
...
- 
-### Required RightScale Roles
- 
-- admin or credential_viewer
 
-### AWS Required Permissions
+## Input Parameters
+
+- Email addresses of the recipients you wish to notify* - A list of email addresses to notify
+- Ignore tags* - ALB/NLB with any of these tags will be ignored
```
Comment (cdwilhelm) [inline security/aws/elb_unencrypted/README.md]: Missing the leading *

File: security/aws/elb_unencrypted/aws_elb_encryption.pt [hunk 14]
```diff
@@ -2,9 +2,16 @@ name "AWS Unencrypted ELB Listeners (ALB/NLB)"
 rs_pt_ver 20180301
 type "policy"
 short_description "Report any AWS App/Network Load Balancers w/Internet-facing Unencrypted Listeners. \n See the [README](https://github.com/rightscale/policy_templates/tree/master/security/aws/elb_unencrypted) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
-long_description "Version: 1.0"
+long_description ""
 category "Security"
 severity "medium"
+tenancy "single"
```
Comment (cdwilhelm) [inline security/aws/elb_unencrypted/aws_elb_encryption.pt]: remove tenancy

File: security/aws/rds_unencrypted/AWS_Unencrypted_RDS_Instances.pt [hunk 15]
```diff
@@ -2,9 +2,16 @@ name "AWS Unencrypted RDS Instances"
 rs_pt_ver 20180301
 type "policy"
 short_description "Report any Relational Database Service (RDS) instances that are unencrypted. \n See the [README](https://github.com/rightscale/policy_templates/tree/master/security/aws/rds_unencrypted) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
-long_description "Version: 1.1"
+long_description ""
 category "Security"
 severity "medium"
+tenancy "single"
```
Comment (cdwilhelm) [inline security/aws/rds_unencrypted/AWS_Unencrypted_RDS_Instances.pt]: Remove tenancy

File: security/aws/rds_unencrypted/README.md [hunk 16]
```diff
@@ -1,70 +1,73 @@
...
-
-### Enable delete action
+## Functional Details
+
+When a Unencrypted RDS instance is detected, an email action is triggered automatically to notify the specified users of the incident. Users then have the option to delete the RDS instance after manual approval if needed.
+
+## Input Parameters
+
+- Email addresses of the recipients you wish to notify* - A list of email addresses to notify
+- Ignore tags* - RDS instances with any of these tags will be ignored
```
Comment (cdwilhelm) [inline security/aws/rds_unencrypted/README.md]: Missing leading *

File: security/aws/unencrypted_s3_buckets/AWS_Unencrypted_S3_Buckets.pt [hunk 17]
```diff
@@ -1,10 +1,17 @@
 name "AWS Unencrypted S3 Buckets"
 rs_pt_ver 20180301
 type "policy"
-short_description "Report any S3 buckets in AWS that are unencrypted and provide the option to set the default encryption after approval. \n See the [README](https://github.com/rightscale/policy_templates/tree/master/security/aws/unencrypted_s3_buckets) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
-long_description "Version: 1.0"
+short_description "Report any S3 buckets in AWS that are unencrypted and provide the option to set the default encryption after approval. \n See the [README](https://github.com/flexera/policy_templates/tree/master/security/aws/unencrypted_s3_buckets) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
+long_description ""
 category "Security"
 severity "low"
+tenancy "single"
```
Comment (cdwilhelm) [inline security/aws/unencrypted_s3_buckets/AWS_Unencrypted_S3_Buckets.pt]: remove tenancy

File: security/aws/unencrypted_s3_buckets/AWS_Unencrypted_S3_Buckets.pt [hunk 18]
```diff
@@ -35,11 +42,11 @@ end
...
 
-auth "auth_aws", type: "aws" do
-  version "4"
-  service "s3"
-  access_key cred("AWS_ACCESS_KEY_ID")
-  secret_key cred("AWS_SECRET_ACCESS_KEY")
+credentials "auth_aws" do
+  schemes "aws"
+  label "AWS"
+  description "Select the AWS IAM Credential from the list"
```
Comment (cdwilhelm) [inline security/aws/unencrypted_s3_buckets/AWS_Unencrypted_S3_Buckets.pt]: remove IAM

**Comment by KTebrewal** [comment]
@cdwilhelm - I will address these comments once i am done with an integration priority task. Anyways this PR was still under progress. I still had to make changes. I missed out leaving a comment. Will request a review once it is updated.

**Comment by KTebrewal** [comment]
@cdwilhelm - While this PR is still in progress, i have two doubts here.
1) I am getting errors in Danger file like - Policy Template path should be lowercase. security/aws/unencrypted_s3_buckets/AWS_Unencrypted_S3_Buckets.pt
To change the Policy name cases, i deleted the files from my repository and renamed in lower case, but it still does not change the cases in policy name. How can we make the changes here?
2) You had asked to remove IAM for only a few policies. It should be for only those or i should apply that for all the policies?

**Comment by cdwilhelm** [comment]
@KTebrewal   When referencing IAM for the credentials IAM should be removed in all cases as the policies should be able to support AWS IAM user and AWS STS (cross-account).  

Regarding the filename, I'm guessing you are windows and the default file naming is case insensitive.  You probably need to disable case insensitive settings on your system.  You can try to google how to change file name case on windows.

**Comment by cdwilhelm** [comment]
@KTebrewal   we found the AWS policies credential schemes requires "aws","aws_sts"  Can you please update the policies with the "aws","aws_sts" schemes.  thanks

**Comment by KTebrewal** [comment]
@cdwilhelm - I have addressed all the comments. There is one issue with the credentials -
i have added schema for AWS IAM and AWS STS credentials, but when i test it in UI, i only see the credentials for AWS IAM.
image

**Pull request merged**